src/Psalm/Internal/Provider/ReturnTypeProvider/ParseUrlReturnTypeProvider.php

```diff
@@ -110,22 +109,29 @@ public static function getFunctionReturnType(FunctionReturnTypeProviderEvent $ev
 
                         return $nullable_falsable_int;
                     }
+
+                    if ($component_type->isSingleIntLiteral()) {
+                        $component_type_type = $component_type->getSingleIntLiteral();
+                        $is_default_component = $component_type_type->value <= 1;
```

Comment:
The snippet https://3v4l.org/uD6RR seems to show that any value inferior or equal to -1 returns an array. Values between 0 and 7 returns null and superior or equal to 8 returns an error